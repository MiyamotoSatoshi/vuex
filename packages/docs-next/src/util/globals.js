const IN_BROWSER = typeof window !== 'undefined'
const IS_PROD = process.env.NODE_ENV === 'production'

module.exports = {
  IN_BROWSER,
  IS_PROD,
}
