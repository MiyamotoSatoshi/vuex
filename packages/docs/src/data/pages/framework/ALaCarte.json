{
  "title": "header",
  "titleText": "headerText",
  "children": [
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "importHeader"
        },
        {
          "type": "alert",
          "lang": "webpackVersionWarning",
          "value": "warning"
        },
        {
          "type": "text",
          "lang": "importText1"
        },
        {
          "type": "markup",
          "value": "js_a_la_carte"
        },
        {
          "type": "alert",
          "lang": "optionsDirectivesNotice",
          "value": "info"
        },
        {
          "type": "text",
          "lang": "importText2"
        },
        {
          "type": "markup",
          "value": "js_a_la_carte_destructuring"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "styleHeader"
        },
        {
          "type": "text",
          "lang": "styleText1"
        },
        {
          "type": "markup",
          "value": "bash_yarn_install_stylus"
        },
        {
          "type": "text",
          "lang": "styleText2"
        },
        {
          "type": "markup",
          "value": "js_import_stylus"
        },
        {
          "type": "alert",
          "lang": "stylusDetailNotice",
          "value": "info"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "loaderHeader"
        },
        {
          "type": "text",
          "lang": "loaderText1"
        },
        {
          "type": "markup",
          "value": "js_plugin_vuetify"
        },
        {
          "type": "markup",
          "value": "js_a_la_carte_webpack"
        },
        {
          "type": "text",
          "lang": "loaderText2"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "loaderLimitationsHeader"
        },
        {
          "type": "text",
          "lang": "loaderLimitationsText1"
        },
        {
          "type": "heading",
          "lang": "loaderLimitationsDynamicHeader"
        },
        {
          "type": "text",
          "lang": "loaderLimitationsDynamicText1"
        },
        {
          "type": "markup",
          "value": "html_a_la_carte_string_tag"
        },
        {
          "type": "markup",
          "value": "js_a_la_carte_manual"
        },
        {
          "type": "text",
          "lang": "loaderLimitationsDynamicText2"
        },
        {
          "type": "markup",
          "value": "html_a_la_carte_dynamic_component"
        },
        {
          "type": "heading",
          "lang": "loaderLimitationsFunctionalHeader"
        },
        {
          "type": "text",
          "lang": "loaderLimitationsFunctionalText1"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "up-next",
          "value": [
            "components/api-explorer"
          ]
        }
      ]
    }
  ]
}
