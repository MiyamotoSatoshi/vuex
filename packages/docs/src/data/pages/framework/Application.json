{
  "title": "header",
  "titleText": "headerText",
  "children": [
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "header2"
        },
        {
          "type": "text",
          "lang": "headerText1"
        },
        {
          "type": "markup",
          "value": "ts_application_service_structure"
        },
        {
          "type": "text",
          "lang": "headerText2"
        },
        {
          "type": "markup",
          "value": "js_application_service_property_reference"
        },
        {
          "type": "heading",
          "lang": "appHeader"
        },
        {
          "type": "text",
          "lang": "appText"
        },
        {
          "type": "alert",
          "lang": "alert1",
          "value": "error"
        },
        {
          "type": "heading",
          "lang": "markupHeader"
        },
        {
          "type": "text",
          "lang": "markupText"
        },
        {
          "type": "text",
          "lang": "markupText2"
        },
        {
          "type": "markup",
          "value": "html_default_markup"
        },
        {
          "type": "alert",
          "lang": "alert2",
          "value": "info"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "up-next",
          "value": [
            "themes/premium",
            "framework/colors",
            "framework/theme"
          ]
        }
      ]
    }
  ]
}
