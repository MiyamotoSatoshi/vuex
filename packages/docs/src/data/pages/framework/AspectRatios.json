{
  "title": "header",
  "titleText": "headerText",
  "mdLink": "https://material.io/design/layout/spacing-methods.html#containers-aspect-ratios",
  "children": [
    {
      "type": "section",
      "children": [
        {
          "type": "usage",
          "value": "ratios"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "api",
          "value": [
            "v-responsive"
          ]
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "up-next",
          "value": [
            "components/cards",
            "components/sheets",
            "components/images"
          ]
        }
      ]
    }
  ]
}
