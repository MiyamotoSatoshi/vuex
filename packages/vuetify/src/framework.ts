import { install } from './install'

// Types
import {
  VuetifyService,
  VuetifyServiceContract,
} from 'vuetify/types/services'
import { VuetifyPreset } from 'vuetify/types/presets'
import Vue from 'vue'

// Services
import * as services from './services'

// Styles
import './styles/main.sass'

export default class Vuetify {
  static install = install

  static installed = false

  static version = __VUETIFY_VERSION__

  framework: Record<string, VuetifyServiceContract> = {}

  installed: string[] = []

  preset: Partial<VuetifyPreset> = {}

  constructor (preset: Partial<VuetifyPreset> = {}) {
    this.preset = preset

    this.use(services.Application)
    this.use(services.Breakpoint)
    this.use(services.Goto)
    this.use(services.Icons)
    this.use(services.Lang)
    this.use(services.Theme)
  }

  // Called on the new vuetify instance
  // bootstrap in install beforeCreate
  // Exposes ssrContext if available
  init (root: Vue, ssrContext?: object) {
    this.installed.forEach(property => {
      const service = this.framework[property]
      service.framework = this.framework

      service.init(root, ssrContext)
    })

    // rtl is not installed and
    // will never be called by
    // the init process
    this.framework.rtl = Boolean(this.preset.rtl) as any
  }

  // Instantiate a VuetifyService
  use (Service: VuetifyService) {
    const property = Service.property

    if (this.installed.includes(property)) return

    this.framework[property] = new Service(this.preset[property])
    this.installed.push(property)
  }
}
