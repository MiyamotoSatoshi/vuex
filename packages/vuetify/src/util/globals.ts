export const IN_BROWSER = typeof window !== 'undefined'
export const IS_DEBUG = process.env.DEBUG === 'true'
export const IS_PROD = process.env.NODE_ENV === 'production'
export const SUPPORTS_INTERSECTION = IN_BROWSER && 'IntersectionObserver' in window
