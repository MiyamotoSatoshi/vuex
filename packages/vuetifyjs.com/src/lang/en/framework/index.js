import Icons from './Icons.json'
import Internationalization from './Internationalization.json'

export default {
  Icons,
  Internationalization
}
