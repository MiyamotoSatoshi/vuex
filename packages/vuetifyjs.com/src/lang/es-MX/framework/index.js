// import Internationalization from './Internationalization.json'

export default {
  // Internationalization
}
