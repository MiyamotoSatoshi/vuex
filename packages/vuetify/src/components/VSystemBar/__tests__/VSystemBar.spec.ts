// Components
import { VSystemBar } from '..'

// Utilities
import { createVuetify } from '@/framework'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from '@jest/globals'

describe('VSystemBar', () => {
  const vuetify = createVuetify()

  function mountFunction (options = {}) {
    return mount(VSystemBar, {
      global: { plugins: [vuetify] },
      ...options,
    })
  }

  it('should match a snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
