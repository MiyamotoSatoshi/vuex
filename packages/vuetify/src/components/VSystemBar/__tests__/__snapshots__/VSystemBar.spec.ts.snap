// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSystemBar should match a snapshot 1`] = `
<div class="v-system-bar v-theme--light">
</div>
`;
